src/main.rs

```diff
@@ -30,9 +35,21 @@ fn word_wrap(paragraph: &str, line_length: usize) -> Vec<String> {
 	result
 }
 
-fn main() {
+fn main() -> Result<()> {
+	color_eyre::install()?;
+
 	let args = Cli::parse();
-	let mut lines = word_wrap(&args.message, 45);
+	let (cols, _) = terminal::size()?;
+
+	let width = match args.width {
+		Some(w) => match u16::from(w) >= cols - 5 {
+			true => cols - 5,
+			false => w.into(),
+		},
+		None => 45,
+	};
```

Comment:
```suggestion
	let width = args.width.unwrap_or(45).min(cols.saturating_sub(5));
```

Reply:
Here we use `saturating_sub` to prevent the app from crashing, when the user terminal is small enough (less than 5 columns)

Reply:
That seems like different code... it was checking if a) `args.width` was defined/existed and b) making sure the width wasn't longer than the size of the terminal itself. Maybe...
```suggestion
	let mut width = args.width.unwrap_or(45).min(cols.saturating_sub(5));
	if width > cols - 5 {
		width = cols - 5;
	};
```

Reply:
I think this is how the code suggested above works...

Reply:
Is there a `.max()` method? That's what my code was doing and yours isn't. What happens when the users sets the width to 5000 and the terminal isn't that long? Lines will wrap and it won't look right, so we have to set it to a max of the width of the terminal (-5 for aesthetics) in that case.

Reply:
Just try it out!

![1](https://github.com/uncenter/kittysay/assets/17523360/90b7ace4-5d36-43e0-a426-23cb57da9e7d)

Reply:
Ooh I see. ChatGPT helped explain and it makes sense now ahah.